client/components/AllPuzzles.js

```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import {connect} from "react-redux"
+import {fetchAllPuzzles} from "../store/puzzles"
+
+
+export class AllPuzzles extends React.Component{
+  constructor(props){
+    super(props)
+  }
+  componentDidMount(){
+    this.props.fetchAllPuzzles()
+
+  }
+  render(){
+    allPuzzles= this.props.puzzle
```

Comment:
has allPuzzles been definied yet? I think youll need to do 
```suggestion
    let allPuzzles= this.props.puzzle
```